common/models/wordbreakers/src/dict/index.ts

```diff
@@ -0,0 +1,362 @@
+import { LazySpan, positionAfter } from "../default/index.js";
+
+// Based on the MIN_KEYSTROKE_PROBABILITY penalty used by the lm-worker.
+const CHAR_SKIP_PENALTY = -Math.log2(.0001);
+
+// const DEFAULT_PARAMS = {
+
+// }
+
+export function splitOnWhitespace(text: string): Span[] {
+  const sections: Span[] = [];
+
+  let start = 0;
+
+  // Surrogate pairs will never overlap \u0020, so we don't need to be
+  // surrogate-pair aware here.
+  for(let index = 0; index < text.length + 1; index++) {
+    let char = (index == text.length ? ' ' : text.charAt(index));
+    if(char.match(/\s/) || char == '\u200c') {
+      if(start !== undefined) {
+        sections.push(new LazySpan(text, start, index));
+        start = undefined; // we do not emit whitespace tokens here.
+      }
+    } else if(start === undefined) {
+      start = index;
+    }
+  }
+
+  return sections;
+}
+
+/**
+ * Splits a string into its constituent codepoints.
+ *
+ * BMP chars will be placed within single-char strings; non-BMP chars will
+ * be represented by a single string with their surrogate-chars paired.
+ * @param text
+ * @returns
+ */
+export function splitOnCodepoints(text: string): string[] {
+  const splayed: string[] = [];
+  let tailIndex = 0;
+  let end = text.length;
+  while(tailIndex != end) {
+    const startIndex = tailIndex;
+    tailIndex = positionAfter(text, tailIndex);
+    splayed.push(text.substring(startIndex, tailIndex));
+  }
+
+  return splayed;
```

Comment:
```suggestion
  return [...text];
```

Reply:
... does this actually keep surrogate pairs together when iterating in that manner?  That's the whole point of this setup:  I want both halves of SMP chars to land within a single entry, not two separate ones.

Reply:
![image](https://github.com/keymanapp/keyman/assets/4498365/1dca0710-ce78-4332-bd3c-5b7f3067be3c)

Simple example in Dev Console:
```
> [...'abc🚗def']
['a', 'b', 'c', '🚗', 'd', 'e', 'f']
```

Reply:
'tis truly wonderful

Reply:
The last worry... that spread operator is ES6+, right?  That can't land as-is while we're using ES5 artifacts - the array-style spread is the one operator I couldn't ever get working right for down-compilation when using common `tslib` helpers.

Of course, if we're moving on anyway for 18+, that won't be an issue.

Reply:
Spread operator is apparently Chrome 46+, so I think that'll be okay. Everything else is ok.

https://caniuse.com/mdn-javascript_operators_spread_spread_in_arrays

Reply:
https://github.com/keymanapp/keyman/pull/10973/files#diff-8d3e25816d2ec6ee7fa737b282e517dde30b1e5ab6adbccb846572caf2ceecc4R246 isn't this also a spread type operator? maybe it landed at a different time though.

Reply:
this other line https://github.com/keymanapp/keyman/pull/10973/files#diff-8d3e25816d2ec6ee7fa737b282e517dde30b1e5ab6adbccb846572caf2ceecc4R246
... uses spread in an object literal, which is from Chrome 60+.. https://caniuse.com/mdn-javascript_operators_spread_spread_in_object_literals

Reply:
Object-spread's down-compile helperis is different from the iterator-spread helper.